### PR TITLE
Add the actual implementation of txDisableRecipe (PYL #60)

### PR DESCRIPTION
**Pull request by Tiny-Angel305**

### CODE REVIEW

File: lib/src/features/ipc/handlers/get_cookbooks_handler.dart [hunk 1]
```diff
@@ -10,9 +12,10 @@ class GetCookbooksHandler implements IPCHandler{
     final defaultResponse = SDKIPCResponse<Cookbook>(success: response.success, action: response.action, data: Cookbook.create(), error: response.error, errorCode: response.errorCode);
     try {
       if (response.success) {
-        defaultResponse.data =  Cookbook.create()..mergeFromProto3Json(response);
+        defaultResponse.data =  Cookbook.create()..mergeFromProto3Json(jsonDecode(response.data));
       }
     } on  Exception catch (_){
```
Comment (mustafapylons) [inline lib/src/features/ipc/handlers/get_cookbooks_handler.dart:17]: Is it possible to catch a more specific exception here?

**Review by mustafapylons — COMMENTED** [review]
See comment about default exception. Otherwise good, clean code.

File: lib/src/features/ipc/handlers/get_cookbooks_handler.dart (hunk 1 above)
Comment (Tiny-Angel305) [reply lib/src/features/ipc/handlers/get_cookbooks_handler.dart:17]: @kjawadDeveloper written codes.

Comment (kjawadDeveloper1) [reply lib/src/features/ipc/handlers/get_cookbooks_handler.dart:17]: Yes

**Comment by afti-githobo** [comment]
The code here looks fine, but I want to raise a design question, not really so much for @Tiny-Angel305 as in general: is there any compelling reason not to remove the separate enable/disable recipe calls entirely?

Background here - enable/disable were at _one point_ separate tx messages that had to be created, so having the API reflect that made sense. But that's not actually the case on the backend any more! Recipes just have an `Enabled` field. Which means UpdateRecipe can just... update that. Both of these calls are entirely redundant.

To be clear: I should have been more explicit about my thinking and the implications of everything when, during the cleanup push, I originally stubbed the enable/disable implementations. That's a communication issue and it's on me.

**Comment by mustafapylons** [comment]
@afti-githobo  good points. Would there be any use case where we enable it and at some point, we might decide to disable it, or all will be done by keep updating the enabled field? If "enabled" field is all that's necessary, you are right. After these PRs, we have to clean them up and not have them anymore.

**Comment by Tiny-Angel305** [comment]
> The code here looks fine, but I want to raise a design question, not really so much for @Tiny-Angel305 as in general: is there any compelling reason not to remove the separate enable/disable recipe calls entirely?
> 
> Background here - enable/disable were at _one point_ separate tx messages that had to be created, so having the API reflect that made sense. But that's not actually the case on the backend any more! Recipes just have an `Enabled` field. Which means UpdateRecipe can just... update that. Both of these calls are entirely redundant.
> 
> To be clear: I should have been more explicit about my thinking and the implications of everything when, during the cleanup push, I originally stubbed the enable/disable implementations. That's a communication issue and it's on me.

I saw your comments saying `TODO: remove this`. But what I thought was, having also `enable`/`disable` apis aside `update` wouldn't affect negatively. They are actually nothing but calling `update`.

**Comment by afti-githobo** [comment]
@mustafapylons that use case can be satisfied by updating the enabled field, and since enable/disable operations are now internally just a recipe update _anyway_ you don't even avoid bumping the version by using the enable/disable calls.

If that's the case, we can go ahead and merge this PR for the misc. cleanup, and I'll go ahead and put one out to actually fully strip the calls out today?

@Tiny-Angel305 yeah, it doesn't exactly _hurt_ anything, but they're redundant (b/c they are just creating updaterecipe txs) and in general I think it'd be cleaner and more elegant to have as few API calls as possible. more code = more potential Problems

**Comment by mustafapylons** [comment]
@afti-githobo  based on the discussion, do you want to create a separate PR for the cleanup as an action item for you? 
Also, in the future, can we have this discussion through a meeting in discord etc. or even through JIRA.

**Comment by afti-githobo** [comment]
@mustafapylons yeah, that sounds fine.

and yeah, we should've had this conversation earlier and in a different setting; sorry about that. will try to be more on top of it in future.

**Comment by Tiny-Angel305** [comment]
Can I close this PR without merging? Ah, this contains Jawad's commit as well. I think it should be merged asap as it's a fix for a build error.

**Comment by mustafapylons** [comment]
It can be merged at this point with the caveat that @afti-githobo will have a separate PR.

**Comment by afti-githobo** [comment]
i'm good w/ that @mustafapylons @Tiny-Angel305, so let's merge it

**Pull request merged**